File: App/DesignSystem/ISG/NovaDocumentWizardScreen.swift
import SwiftUI
import UniformTypeIdentifiers
import CryptoKit

struct NovaWizardWorkplace: Identifiable {
    let id: UUID
    let name: String
}
private struct NovaWizardFile: FileDocument {
    static let readableContentTypes: [UTType] = [.data]
    var data: Data
    init(data: Data) { self.data = data }
    init(configuration: ReadConfiguration) throws { data = configuration.file.regularFileContents ?? Data() }
    func fileWrapper(configuration: WriteConfiguration) throws -> FileWrapper { FileWrapper(regularFileWithContents: data) }
}

/// A single accordion page, matching EducationEditor. Scope is optional at every step.
struct NovaDocumentWizardScreen: View {
    let domain: String
    let companiesSource: () async throws -> [NovaAnalysisCompanyOption]
    let workplacesSource: (UUID) async throws -> [NovaWizardWorkplace]
    let files: NovaFileLibraryClient
    var initialCompany: UUID?
    let onBack: () -> Void

    @State private var runtime: NovaDocumentWizardRuntime?
    @State private var companies: [NovaAnalysisCompanyOption] = []
    @State private var workplaces: [NovaWizardWorkplace] = []
    @State private var company: UUID?
    @State private var workplace: UUID?
    @State private var selections: [String: Set<String>] = [:]
    @State private var states: [String: String] = [:]
    @State private var method = "fk"
    @State private var preset = "standard"
    @State private var step = 0
    @State private var search = ""
    @State private var shown = 12
    @State private var preview: NovaWizardPreview?
    @State private var message: String?
    @State private var scopeMessage: String?
    @State private var busy = false
    @State private var archivedHash: String?
    @State private var exportFile = NovaWizardFile(data: Data())
    @State private var exportName = "ISGADA.docx"
    @State private var exporting = false

    private var questions: [NovaWizardQuestion] { (runtime?.questions ?? []).filter { domain == "risk" || !["method", "preset"].contains($0.field) } }
    private var title: String { domain == "risk" ? "Risk Analizi Sihirbazı" : "Acil Durum Planı Sihirbazı" }
    var body: some View {
        NovaPageSurface(onEdgeBack: onBack) {
            ScrollViewReader { scroll in
                ScrollView {
                    VStack(alignment: .leading, spacing: 12) {
                        NovaPageHeading(title: title, onBack: onBack)
                        if let runtime {
                            NovaHelpHint(text: RDLocalization.string("localizable.nova.document.wizard.screen.kapsaminizi.secin.taslagi.olusturun.ve.word.exce.eddf3ec3", table: .localizable, fallback: "Kapsamınızı seçin, taslağı oluşturun ve Word, Excel veya PDF olarak indirin. Firma seçimi isteğe bağlıdır."))
                            ForEach(Array(questions.enumerated()), id: \.element.id) { index, q in
                                NovaCard(padding: 14) {
                                    VStack(alignment: .leading, spacing: 12) {
                                        Button { move(to: index) } label: {
                                            HStack(alignment: .top) {
                                                Text("\(index + 1)").font(NovaFont.font(.meta))
                                                VStack(alignment: .leading, spacing: 4) {
                                                    NovaText(text: q.text_tr, style: .cardTitle)
                                                    if step != index { NovaText(text: answerSummary(q.field), style: .metaQuiet) }
                                                }
                                                Spacer(minLength: 4)
                                                Image(systemName: step == index ? "chevron.up" : "chevron.down")
                                            }.contentShape(Rectangle())
                                        }.buttonStyle(.plain).accessibilityIdentifier("wizard.step.\(q.field)")
                                        if step == index {
                                            NovaText(text: q.help_tr, style: .meta)
                                            question(q, runtime: runtime)
                                            HStack {
                                                if index > 0 { NovaButton(label: "Geri", symbol: "chevron.left", variant: .surface) { move(to: index - 1) } }
                                                Spacer(minLength: 0)
                                                NovaButton(label: index == questions.count - 1 ? "Taslak oluştur" : "Devam", symbol: "chevron.right", isEnabled: !busy) {
                                                    if index == questions.count - 1 { generate() } else { move(to: index + 1) }
                                                }.accessibilityIdentifier("wizard.next")
                                            }
                                        }
                                    }
                                }.id(index).disabled(busy)
                            }
                            if let preview { result(preview).id(questions.count) }
                        } else if message == nil { ProgressView().frame(maxWidth: .infinity) }
                        if let message { NovaText(text: message, style: .meta) }
                    }.padding(.horizontal, 20).padding(.top, 12).padding(.bottom, 24 + novaTabBarInset)
                }
                .onChange(of: step) { value in
                    withAnimation(.easeInOut(duration: 0.2)) { scroll.scrollTo(value, anchor: .top) }
                }
            }
        }
        .task {
            do { runtime = try NovaDocumentWizardRuntime() } catch { message = error.localizedDescription }
            do {
                companies = try await companiesSource()
                if let initialCompany, companies.contains(where: { $0.id == initialCompany }) { company = initialCompany }
            } catch { scopeMessage = "Firmalar yüklenemedi. Firma seçmeden devam edebilir veya sayfayı yeniden açabilirsiniz." }
        }
        .task(id: company) { await loadWorkplaces() }
        .fileExporter(isPresented: $exporting, document: exportFile,
            contentType: UTType(filenameExtension: (exportName as NSString).pathExtension) ?? .data,
            defaultFilename: exportName) { result in
                if case .failure = result { message = "Dosya kaydedilemedi. Tekrar deneyebilirsiniz." }
                exportFile = .init(data: Data())
            }
    }

    @ViewBuilder private func question(_ q: NovaWizardQuestion, runtime: NovaDocumentWizardRuntime) -> some View {
        if q.field == "scope" {
            Picker("Firma", selection: $company) {
                Text(RDLocalization.string("localizable.nova.document.wizard.screen.firma.secmeden.devam.et.36ce6fb4", table: .localizable, fallback: "Firma seçmeden devam et")).tag(nil as UUID?)
                ForEach(companies) { Text($0.name).tag(Optional($0.id)) }
            }.pickerStyle(.menu).accessibilityIdentifier("wizard.company")
                .onChange(of: company) { _ in workplace = nil; workplaces = []; changed() }
            if !workplaces.isEmpty {
                Picker(RDLocalization.string("localizable.nova.document.wizard.screen.isyeri.198609fd", table: .localizable, fallback: "İşyeri"), selection: $workplace) {
                    Text(RDLocalization.string("localizable.nova.document.wizard.screen.isyeri.secmeden.devam.et.7f870416", table: .localizable, fallback: "İşyeri seçmeden devam et")).tag(nil as UUID?)
                    ForEach(workplaces) { Text($0.name).tag(Optional($0.id)) }
                }.pickerStyle(.menu).accessibilityIdentifier("wizard.workplace")
                    .onChange(of: workplace) { _ in changed() }
            }
            if let scopeMessage { NovaText(text: scopeMessage, style: .metaQuiet) }
        } else if q.field == "method" || q.field == "preset" {
            let selected = q.field == "method" ? method : preset
            ForEach((runtime.options[q.field] ?? [:]).keys.sorted(), id: \.self) { key in
                option(runtime.options[q.field]?[key] ?? key, selected: selected == key) {
                    if q.field == "method" { method = key } else { preset = key }; changed()
                }
            }
        } else {
            let choices = runtime.choices[q.field] ?? []
            NovaAnalysisSearchField(text: $search, placeholder: RDLocalization.string("localizable.nova.document.wizard.screen.secenek.ara.7bf34387", table: .localizable, fallback: "Seçenek ara"), identifier: "wizard.search.\(q.field)")
                .onChange(of: search) { _ in shown = 12 }
            let selected = selections[q.field] ?? []
            let filtered = choices.filter { search.isEmpty || ([$0.label] + $0.aliases).contains { searchKey($0).contains(searchKey(search)) } }
                .sorted { a, b in selected.contains(a.id) != selected.contains(b.id) ? selected.contains(a.id) : a.label.localizedStandardCompare(b.label) == .orderedAscending }
            ForEach(Array(filtered.prefix(shown))) { choice in
                option(choice.label, selected: selected.contains(choice.id)) {
                    var next = selected
                    if next.contains(choice.id) { next.remove(choice.id) }
                    else if next.count < 100 { next.insert(choice.id) }
                    else { message = "Bu başlıkta en fazla 100 seçenek seçebilirsiniz."; return }
                    selections[q.field] = next; states[q.field] = next.isEmpty ? "unknown" : "selected"; changed()
                }
            }
            if filtered.isEmpty { NovaText(text: RDLocalization.string("localizable.nova.document.wizard.screen.eslesen.secenek.yok.diger.sorulardan.kapsam.ekle.cc0f299e", table: .localizable, fallback: "Eşleşen seçenek yok. Diğer sorulardan kapsam ekleyebilirsiniz."), style: .metaQuiet) }
            if filtered.count > shown { NovaButton(label: RDLocalization.string("localizable.nova.document.wizard.screen.daha.fazla.goster.956cb4d1", table: .localizable, fallback: "Daha fazla göster"), symbol: "chevron.down", variant: .surface) { shown += 20 } }
            option("Yok", selected: states[q.field] == "none") { selections[q.field] = []; states[q.field] = "none"; changed() }
            option("Bilmiyorum / sonra tamamlayacağım", selected: (states[q.field] ?? "unknown") == "unknown") { selections[q.field] = []; states[q.field] = "unknown"; changed() }
            NovaText(text: RDLocalization.format("localizable.nova.document.wizard.screen.1.secenek.secildi.fd69b1e3", table: .localizable, fallback: "%1$@ seçenek seçildi", arguments: [String(describing: selected.count)]), style: .metaQuiet)
        }
    }
    private func option(_ label: String, selected: Bool, action: @escaping () -> Void) -> some View {
        Button(action: action) {
            HStack(alignment: .top, spacing: 10) {
                Image(systemName: selected ? "checkmark.circle.fill" : "circle")
                NovaText(text: label, style: .body)
                Spacer(minLength: 0)
            }.padding(.vertical, 8).frame(minHeight: 44).contentShape(Rectangle())
        }.buttonStyle(.plain).accessibilityAddTraits(selected ? .isSelected : [])
    }
    private func searchKey(_ value: String) -> String {
        value.folding(options: [.diacriticInsensitive, .caseInsensitive], locale: Locale(identifier: "tr_TR"))
            .replacingOccurrences(of: "ı", with: "i")
    }
    private func move(to index: Int) { step = index; search = ""; shown = 12 }
    private func changed() { preview = nil; archivedHash = nil; message = nil }
    private func answerSummary(_ field: String) -> String {
        if field == "scope" { return companies.first { $0.id == company }?.name ?? "Firma seçmeden" }
        if field == "method" { return runtime?.options[field]?[method] ?? method }
        if field == "preset" { return runtime?.options[field]?[preset] ?? preset }
        if states[field] == "none" { return "Yok" }
        let count = selections[field]?.count ?? 0
        return count == 0 ? "Belirtilmedi" : "\(count) seçenek"
    }
    private func loadWorkplaces() async {
        workplace = nil; workplaces = []
        guard let selected = company else { return }
        do {
            let answer = try await workplacesSource(selected)
            guard !Task.isCancelled, company == selected else { return }
            workplaces = answer; scopeMessage = nil
        } catch {
            guard !Task.isCancelled, company == selected else { return }
            scopeMessage = "İşyeri listesi yüklenemedi. İşyeri seçmeden devam edebilirsiniz."
        }
    }
    private func generate() {
        guard let runtime else { return }
        var answers: [String: Any] = ["method": method, "preset": preset,
            "scope": ["company_id": company?.uuidString ?? "", "workplace_id": workplace?.uuidString ?? "",
                "company_name": companies.first { $0.id == company }?.name ?? "", "workplace_name": workplaces.first { $0.id == workplace }?.name ?? ""]]
        for key in runtime.choices.keys { answers[key] = ["state": states[key] ?? "unknown", "ids": Array(selections[key] ?? []).sorted()] }
        do {
            preview = try runtime.generate(answers: answers, domain: domain); step = questions.count; message = nil
            if domain == "emergency" { NovaForYouOutbox.recordUse("emergency_wizard") }
        }
        catch { message = error.localizedDescription }
    }
    @ViewBuilder private func result(_ preview: NovaWizardPreview) -> some View {
        NovaCard(padding: 16) {
            VStack(alignment: .leading, spacing: 12) {
                NovaText(text: RDLocalization.string("localizable.nova.document.wizard.screen.taslak.hazir.538b19f8", table: .localizable, fallback: "Taslak hazır"), style: .cardTitle)
                NovaText(text: domain == "risk" ? "\(preview.rows.count) risk · \(preview.unresolved.count) kapsamı netleşmemiş konu · \(preview.omitted.count) sınır dışında" : "\(preview.cards.count) acil durum eylem kartı", style: .meta)
                ForEach(preview.input_conflicts, id: \.self) { NovaHelpHint(text: $0) }
                ForEach(["docx", "xlsx", "pdf"], id: \.self) { format in
                    NovaButton(label: ["docx":"Word indir", "xlsx":"Excel indir", "pdf":"PDF indir"][format]!, symbol: "arrow.down.doc", variant: .surface, isEnabled: !busy) {
                        do { guard let runtime else { throw NovaWizardError.unavailable }; let file = try runtime.download(format: format); exportName = file.name; exportFile = .init(data: file.data); exporting = true }
                        catch { message = error.localizedDescription }
                    }
                }
                NovaButton(label: archivedHash == preview.content_sha256 ? "Arşive gönderildi" : "Word dosyasını arşive ekle", symbol: "folder.badge.plus", isEnabled: !busy && archivedHash != preview.content_sha256) {
                    Task { await archive(preview) }
                }
                if busy { ProgressView() }
                DisclosureGroup("Belge içeriği") {
                    if domain == "risk" { ForEach(preview.rows) { NovaText(text: "\($0.id) · \($0.scenario_tr)", style: .meta).padding(.vertical, 4) } }
                    else { ForEach(preview.cards) { NovaText(text: "\($0.id) · \($0.title_tr)", style: .meta).padding(.vertical, 4) } }
                }
                if domain == "risk" && !preview.unresolved.isEmpty {
                    DisclosureGroup("Kapsamı netleşmemiş konular") {
                        ForEach(preview.unresolved) { gap in
                            NovaText(text: "\(gap.title)\nGerekli bilgi: \(gap.missing_labels.joined(separator: "; "))", style: .meta).padding(.vertical, 6)
                        }
                    }
                    NovaText(text: RDLocalization.string("localizable.nova.document.wizard.screen.ilgili.soruya.donup.kapsam.secimini.genisletebil.49c4cc1e", table: .localizable, fallback: "İlgili soruya dönüp kapsam seçimini genişletebilirsiniz. Bu konular belgenin ekinde de yer alır."), style: .metaQuiet)
                }
                if domain == "risk" && !preview.omitted.isEmpty {
                    DisclosureGroup("Satır sınırı dışında kalan konular") { ForEach(preview.omitted) { NovaText(text: "\($0.id) · \($0.title)", style: .meta).padding(.vertical, 4) } }
                }
            }
        }
    }
    private func archive(_ preview: NovaWizardPreview) async {
        guard let runtime, !busy else { return }
        busy = true; defer { busy = false }
        do {
            let download = try runtime.download(format: "docx")
            let draft = NovaFileDraft(title: title + " · Taslak", category: domain == "risk" ? "risk_assessment" : "emergency_plan",
                note: "Sihirbaz taslağı · \(preview.content_sha256)", fileName: download.name, fileExtension: "docx", bytes: download.data.count,
                sha256: SHA256.hash(data: download.data).map { String(format: "%02x", $0) }.joined())
            let entry = try await files.file(company, draft, download.data)
            guard !Task.isCancelled else { return }
            archivedHash = preview.content_sha256
            message = "Dosyalarım: " + NovaFileWords.state(entry.state) + "."
        } catch let error as NovaFileFailure { message = NovaFileScreenWords.failure(error) }
        catch { message = "Dosya arşive eklenemedi. İndirerek kullanabilir veya tekrar deneyebilirsiniz." }
    }
}

File: android/feature/nova/src/main/kotlin/com/riskdetectedan/feature/nova/NovaDocumentWizardScreen.kt
package com.riskdetectedan.feature.nova

import androidx.activity.compose.rememberLauncherForActivityResult
import androidx.activity.result.contract.ActivityResultContracts
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.itemsIndexed
import androidx.compose.foundation.lazy.rememberLazyListState
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.semantics.selected
import androidx.compose.ui.semantics.semantics
import androidx.compose.ui.unit.dp
import com.riskdetectedan.core.data.nova.*
import com.riskdetectedan.core.designsystem.isg.*
import kotlinx.coroutines.CancellationException
import kotlinx.coroutines.launch
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext
import kotlinx.serialization.json.*
import java.util.Locale
import java.text.Normalizer

@Composable
fun NovaDocumentWizardScreen(domain: String, companiesSource: suspend () -> List<NovaCompanyOption>,
                             workplacesSource: suspend (String) -> List<NovaWizardWorkplace>, files: NovaFileClient,
                             initialCompany: String? = null, onBack: () -> Unit) {
    val context = LocalContext.current
    val coroutines = rememberCoroutineScope()
    val list = rememberLazyListState()
    var runtime by remember { mutableStateOf<NovaDocumentWizardRuntime?>(null) }
    var companies by remember { mutableStateOf<List<NovaCompanyOption>>(emptyList()) }
    var workplaces by remember { mutableStateOf<List<NovaWizardWorkplace>>(emptyList()) }
    var company by remember { mutableStateOf<String?>(null) }
    var workplace by remember { mutableStateOf<String?>(null) }
    var selections by remember { mutableStateOf<Map<String, Set<String>>>(emptyMap()) }
    var states by remember { mutableStateOf<Map<String, String>>(emptyMap()) }
    var method by remember { mutableStateOf("fk") }
    var preset by remember { mutableStateOf("standard") }
    var step by remember { mutableIntStateOf(0) }
    var search by remember { mutableStateOf("") }
    var shown by remember { mutableIntStateOf(12) }
    var chooser by remember { mutableStateOf<String?>(null) }
    var preview by remember { mutableStateOf<JsonObject?>(null) }
    var message by remember { mutableStateOf<String?>(null) }
    var scopeMessage by remember { mutableStateOf<String?>(null) }
    var busy by remember { mutableStateOf(false) }
    var archivedHash by remember { mutableStateOf<String?>(null) }
    var download by remember { mutableStateOf<NovaWizardDownload?>(null) }
    var expanded by remember { mutableStateOf(false) }
    fun visibleQuestions(rt: NovaDocumentWizardRuntime) = rt.questions.filter { domain == "risk" || it.field !in listOf("method", "preset") }
    val title = if (domain == "risk") "Risk Analizi Sihirbazı" else "Acil Durum Planı Sihirbazı"
    fun changed() { preview = null; archivedHash = null; message = null }
    fun move(index: Int) { step = index; search = ""; shown = 12; chooser = null }
    val save = rememberLauncherForActivityResult(ActivityResultContracts.CreateDocument("application/octet-stream")) { uri ->
        val captured = download
        if (uri != null && captured != null) coroutines.launch {
            message = try {
                withContext(Dispatchers.IO) { context.contentResolver.openOutputStream(uri)?.use { it.write(captured.bytes) } ?: error("Destination unavailable") }
                "Dosya kaydedildi."
            } catch (e: CancellationException) { throw e } catch (_: Exception) { "Dosya kaydedilemedi. Tekrar deneyebilirsiniz." }
            download = null
        } else download = null
    }
    DisposableEffect(Unit) { onDispose { runtime?.close() } }
    LaunchedEffect(Unit) {
        try { runtime = NovaDocumentWizardRuntime(context); runtime?.ready() }
        catch (e: CancellationException) { throw e } catch (_: Exception) { runtime?.close(); runtime = null; message = "Sihirbaz içeriği yüklenemedi." }
        try {
            companies = companiesSource()
            if (companies.any { it.id == initialCompany }) company = initialCompany
        } catch (e: CancellationException) { throw e } catch (_: Exception) { scopeMessage = "Firmalar yüklenemedi. Firma seçmeden devam edebilirsiniz." }
    }
    LaunchedEffect(company) {
        workplaces = emptyList(); workplace = null
        val selected = company
        if (selected != null) try { workplaces = workplacesSource(selected); scopeMessage = null }
        catch (e: CancellationException) { throw e } catch (_: Exception) { scopeMessage = "İşyeri listesi yüklenemedi. İşyeri seçmeden devam edebilirsiniz." }
    }
    LaunchedEffect(step) { if (runtime != null && list.layoutInfo.totalItemsCount > step + 2) list.animateScrollToItem(step + 2) }
    fun answers(rt: NovaDocumentWizardRuntime) = buildJsonObject {
        put("method", method); put("preset", preset)
        putJsonObject("scope") { put("company_id", company); put("workplace_id", workplace); put("company_name", companies.firstOrNull { it.id == company }?.name.orEmpty()); put("workplace_name", workplaces.firstOrNull { it.id == workplace }?.name.orEmpty()) }
        rt.choices.keys.forEach { key -> putJsonObject(key) { put("state", states[key] ?: "unknown"); put("ids", JsonArray(selections[key].orEmpty().sorted().map(::JsonPrimitive))) } }
    }
    fun generate(rt: NovaDocumentWizardRuntime) {
        coroutines.launch {
            busy = true
            try {
                preview = rt.generate(answers(rt), domain); step = visibleQuestions(rt).size; message = null
                if (domain == "emergency") NovaForYouService.recordUse("emergency_wizard")
            }
            catch (e: CancellationException) { throw e } catch (e: Exception) { message = e.message ?: "Taslak oluşturulamadı." }
            finally { busy = false }
        }
    }
    fun export(rt: NovaDocumentWizardRuntime, format: String) {
        coroutines.launch {
            busy = true
            try { val file = rt.download(format); download = file; save.launch(file.name) }
            catch (e: CancellationException) { throw e } catch (_: Exception) { message = "Dosya oluşturulamadı. Tekrar deneyebilirsiniz." }
            finally { busy = false }
        }
    }
    fun archive(rt: NovaDocumentWizardRuntime, snapshot: JsonObject) {
        coroutines.launch {
            busy = true
            try {
                val file = rt.download("docx")
                val hash = snapshot.wString("content_sha256")
                val draft = NovaFileDraft(title = "$title · Taslak", category = if (domain == "risk") "risk_assessment" else "emergency_plan", note = "Sihirbaz taslağı · $hash", fileName = file.name, fileExtension = "docx", bytes = file.bytes.size, sha256 = NovaFileDraft.sha256(file.bytes))
                val entry = files.file(company, draft, file.bytes)
                archivedHash = hash; message = "Dosyalarım: ${entry.state.title}."
            } catch (e: CancellationException) { throw e } catch (e: Exception) { message = (e as? NovaFileException)?.failure?.message ?: "Dosya arşive eklenemedi. İndirerek kullanabilirsiniz." }
            finally { busy = false }
        }
    }
    val rt = runtime
    LazyColumn(Modifier.fillMaxSize().padding(horizontal = 20.dp), state = list,
        contentPadding = PaddingValues(top = 12.dp, bottom = 24.dp + novaTabBarInset), verticalArrangement = Arrangement.spacedBy(12.dp)) {
        item { NovaListHeading(title, onBack) }
        item { NovaHelpHint("Kapsamınızı seçin, taslağı oluşturun ve Word, Excel veya PDF olarak indirin. Firma seçimi isteğe bağlıdır.") }
        if (rt != null) itemsIndexed(visibleQuestions(rt), key = { _, q -> q.id }) { index, q ->
            NovaCard(Modifier.fillMaxWidth(), padding = 14) {
                Column(verticalArrangement = Arrangement.spacedBy(12.dp)) {
                    Row(Modifier.fillMaxWidth().clickable(enabled = !busy) { move(index) }.testTag("wizard.step.${q.field}"), horizontalArrangement = Arrangement.spacedBy(10.dp)) {
                        NovaText("${index + 1}", style = NovaTypeToken.meta)
                        Column(Modifier.weight(1f)) {
                            NovaText(q.title, style = NovaTypeToken.cardTitle)
                            if (step != index) NovaText(when (q.field) {
                                "scope" -> companies.firstOrNull { it.id == company }?.name ?: "Firma seçmeden"
                                "method" -> rt.options.wObject("method").wString(method)
                                "preset" -> rt.options.wObject("preset").wString(preset)
                                else -> if (states[q.field] == "none") "Yok" else if (selections[q.field].isNullOrEmpty()) "Belirtilmedi" else "${selections[q.field]?.size} seçenek"
                            }, style = NovaTypeToken.metaQuiet)
                        }
                        NovaIcon(if (step == index) "chevron.up" else "chevron.down", 16.dp)
                    }
                    if (step == index) {
                        NovaText(q.help, style = NovaTypeToken.meta)
                        when (q.field) {
                            "scope" -> {
                                NovaChooserButton("Firma", companies.firstOrNull { it.id == company }?.name ?: "Firma seçmeden devam et", "wizard.company", open = chooser == "company") { if (!busy) chooser = if (chooser == "company") null else "company" }
                                if (chooser == "company") NovaChooserPanel(listOf(NovaChooserOption(null, "Firma seçmeden devam et")) + companies.map { NovaChooserOption(it.id, it.name) }, company, "wizard.companies") { if (!busy) { company = it; workplace = null; workplaces = emptyList(); chooser = null; changed() } }
                                if (workplaces.isNotEmpty()) {
                                    NovaChooserButton("İşyeri", workplaces.firstOrNull { it.id == workplace }?.name ?: "İşyeri seçmeden devam et", "wizard.workplace", open = chooser == "workplace") { if (!busy) chooser = if (chooser == "workplace") null else "workplace" }
                                    if (chooser == "workplace") NovaChooserPanel(listOf(NovaChooserOption(null, "İşyeri seçmeden devam et")) + workplaces.map { NovaChooserOption(it.id, it.name) }, workplace, "wizard.workplaces") { if (!busy) { workplace = it; chooser = null; changed() } }
                                }
                                scopeMessage?.let { NovaText(it, style = NovaTypeToken.metaQuiet) }
                            }
                            "method", "preset" -> rt.options.wObject(q.field).toSortedMap().forEach { (key, value) ->
                                WizardOption(value.jsonPrimitive.content, (if (q.field == "method") method else preset) == key, !busy) { if (q.field == "method") method = key else preset = key; changed() }
                            }
                            else -> {
                                NovaSearchCapsule(search, "Seçenek ara", "wizard.search.${q.field}") { search = it; shown = 12 }
                                val selected = selections[q.field].orEmpty()
                                val filtered = rt.choices[q.field].orEmpty().filter { search.isBlank() || (listOf(it.label) + it.aliases).any { word -> wizardSearchKey(word).contains(wizardSearchKey(search)) } }.sortedWith(compareByDescending<NovaWizardChoice> { it.id in selected }.thenBy { it.label })
                                filtered.take(shown).forEach { choice -> WizardOption(choice.label, choice.id in selected, !busy) {
                                    if (selected.size >= 100 && choice.id !in selected) message = "Bu başlıkta en fazla 100 seçenek seçebilirsiniz."
                                    else { val next = if (choice.id in selected) selected - choice.id else selected + choice.id; selections = selections + (q.field to next); states = states + (q.field to if (next.isEmpty()) "unknown" else "selected"); changed() }
                                } }
                                if (filtered.isEmpty()) NovaText("Eşleşen seçenek yok. Diğer sorulardan kapsam ekleyebilirsiniz.", style = NovaTypeToken.metaQuiet)
                                if (filtered.size > shown) NovaButton("Daha fazla göster", { shown += 20 }, symbol = "chevron.down", variant = NovaButtonVariant.Surface)
                                WizardOption("Yok", states[q.field] == "none", !busy) { selections = selections + (q.field to emptySet()); states = states + (q.field to "none"); changed() }
                                WizardOption("Bilmiyorum / sonra tamamlayacağım", (states[q.field] ?: "unknown") == "unknown", !busy) { selections = selections + (q.field to emptySet()); states = states + (q.field to "unknown"); changed() }
                                NovaText("${selected.size} seçenek seçildi", style = NovaTypeToken.metaQuiet)
                            }
                        }
                        Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.spacedBy(10.dp)) {
                            if (index > 0) NovaButton("Geri", { move(index - 1) }, symbol = "chevron.left", variant = NovaButtonVariant.Surface, enabled = !busy)
                            Spacer(Modifier.weight(1f))
                            NovaButton(if (index == visibleQuestions(rt).lastIndex) "Taslak oluştur" else "Devam", { if (index == visibleQuestions(rt).lastIndex) generate(rt) else move(index + 1) }, symbol = "chevron.right", enabled = !busy)
                        }
                    }
                }
            }
        }
        val snapshot = preview
        if (snapshot != null && rt != null) item {
            NovaCard(Modifier.fillMaxWidth(), padding = 16) {
                Column(verticalArrangement = Arrangement.spacedBy(12.dp)) {
                    NovaText("Taslak hazır", style = NovaTypeToken.cardTitle)
                    NovaText(if (domain == "risk") "${snapshot.wArray("rows").size} risk · ${snapshot.wArray("unresolved").size} kapsamı netleşmemiş konu · ${snapshot.wArray("omitted").size} sınır dışında" else "${snapshot.wArray("cards").size} acil durum eylem kartı", style = NovaTypeToken.meta)
                    snapshot.wArray("input_conflicts").forEach { NovaHelpHint(it.jsonPrimitive.content) }
                    listOf("docx" to "Word indir", "xlsx" to "Excel indir", "pdf" to "PDF indir").forEach { (format, label) -> NovaButton(label, { export(rt, format) }, symbol = "arrow.down.doc", variant = NovaButtonVariant.Surface, enabled = !busy) }
                    NovaButton(if (archivedHash == snapshot.wString("content_sha256")) "Arşive gönderildi" else "Word dosyasını arşive ekle", { archive(rt, snapshot) }, symbol = "folder.badge.plus", enabled = !busy && archivedHash != snapshot.wString("content_sha256"))
                    NovaButton(if (expanded) "İçeriği daralt" else "Belge içeriği ve kapsam", { expanded = !expanded }, symbol = if (expanded) "chevron.up" else "chevron.down", variant = NovaButtonVariant.Surface)
                    if (expanded) {
                        snapshot.wArray(if (domain == "risk") "rows" else "cards").forEach { item -> NovaText(item.jsonObject.wString("id") + " · " + item.jsonObject.wString(if (domain == "risk") "scenario_tr" else "title_tr"), style = NovaTypeToken.meta) }
                        if (domain == "risk") {
                            snapshot.wArray("unresolved").forEach { item -> NovaText("Kapsamı netleşmemiş: " + item.jsonObject.wString("title") + "\nGerekli bilgi: " + item.jsonObject.wArray("missing_labels").joinToString("; ") { it.jsonPrimitive.content }, style = NovaTypeToken.meta) }
                            snapshot.wArray("omitted").forEach { NovaText("Sınır dışında: " + it.jsonObject.wString("title"), style = NovaTypeToken.meta) }
                        }
                    }
                }
            }
        }
        if (busy) item { NovaText("Hazırlanıyor…", style = NovaTypeToken.meta) }
        message?.let { item { NovaText(it, style = NovaTypeToken.meta) } }
    }
}

@Composable
private fun WizardOption(label: String, selected: Boolean, enabled: Boolean, onClick: () -> Unit) {
    Row(Modifier.fillMaxWidth().heightIn(min = 44.dp).clickable(enabled = enabled, onClick = onClick).semantics { this.selected = selected }.padding(vertical = 8.dp), horizontalArrangement = Arrangement.spacedBy(10.dp), verticalAlignment = Alignment.Top) {
        NovaIcon(if (selected) "checkmark.circle.fill" else "circle", 20.dp)
        NovaText(label, style = NovaTypeToken.body, modifier = Modifier.weight(1f))
    }
}

private fun wizardSearchKey(value: String) = Normalizer.normalize(value, Normalizer.Form.NFD)
    .replace(Regex("\\p{M}+"), "").lowercase(Locale.ROOT).replace('ı', 'i')
